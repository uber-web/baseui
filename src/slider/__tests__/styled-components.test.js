/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledTrack,
  StyledInnerTrack,
  StyledTick,
  StyledThumb,
  StyledInnerThumb,
  StyledTickBar,
} from '../index.js';

describe('Slider styled components', () => {
  describe('StyledRoot', function() {
    test('StyledRoot', () => {
      const component = shallow(
        <StyledRoot>
          <div />
        </StyledRoot>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledRoot has correct styles',
      );
    });
  });
  describe('StyledTrack', function() {
    test('StyledTrack', () => {
      const component = shallow(
        <StyledTrack $value={[10, 20]}>
          <div />
        </StyledTrack>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledTrack has correct styles',
      );
    });
  });
  describe('StyledInnerTrack', function() {
    test('StyledInnerTrack', () => {
      const component = shallow(
        <StyledInnerTrack $value={[50]}>
          <div />
        </StyledInnerTrack>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledInnerTrack has correct styles',
      );
    });
  });
  describe('StyledThumb', function() {
    test('StyledThumb', () => {
      const component = shallow(<StyledThumb $value={[50]} $thumbIndex={0} />);
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledThumb has correct styles',
      );
    });
    test('StyledThumb left', () => {
      const component = shallow(
        <StyledThumb $value={[50, 70]} $thumbIndex={0} />,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledThumb left range has correct styles',
      );
    });
    test('StyledThumb right', () => {
      const component = shallow(
        <StyledThumb $value={[50, 70]} $thumbIndex={1} />,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledThumb right range has correct styles',
      );
    });
  });
  describe('StyledInnerThumb', function() {
    test('StyledInnerThumb', () => {
      const component = shallow(<StyledInnerThumb />);
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledInnerThumb has correct styles',
      );
    });
  });
  describe('StyledTick', function() {
    test('StyledTick', () => {
      const component = shallow(
        <StyledTick>
          <div />
        </StyledTick>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledTick has correct styles',
      );
    });
  });
  describe('StyledTickBar', function() {
    test('StyledTickBar', () => {
      const component = shallow(
        <StyledTickBar>
          <div />
        </StyledTickBar>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledTickBar has correct styles',
      );
    });
  });
});
