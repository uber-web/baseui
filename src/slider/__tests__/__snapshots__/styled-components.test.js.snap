// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider styled components StyledInnerThumb StyledInnerThumb: StyledInnerThumb has correct styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono600",
  "borderRadius": "2px",
  "height": "8px",
  "width": "2px",
}
`;

exports[`Slider styled components StyledInnerTrack StyledInnerTrack: StyledInnerTrack has correct styles 1`] = `
Object {
  "alignSelf": "center",
  "background": "linear-gradient(to right, $theme.colors.primary 0%, $theme.colors.primary NaN%, $theme.colors.mono400 NaN%, $theme.colors.mono400 100%)",
  "borderRadius": "$theme.borders.radius100",
  "cursor": "inherit",
  "height": "$theme.sizing.scale100",
  "width": "100%",
}
`;

exports[`Slider styled components StyledRoot StyledRoot: StyledRoot has correct styles 1`] = `Object {}`;

exports[`Slider styled components StyledThumb StyledThumb left: StyledThumb left range has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomLeftRadius": "4px",
  "borderBottomRightRadius": "1px",
  "borderColor": "$theme.colors.mono400",
  "borderStyle": "solid",
  "borderTopLeftRadius": "4px",
  "borderTopRightRadius": "1px",
  "borderWidth": "1px",
  "boxShadow": "0px 1px 4px rgba(0, 0, 0, 0.12)",
  "cursor": "inherit",
  "display": "flex",
  "height": "24px",
  "justifyContent": "center",
  "width": "12px",
}
`;

exports[`Slider styled components StyledThumb StyledThumb right: StyledThumb right range has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomLeftRadius": "1px",
  "borderBottomRightRadius": "4px",
  "borderColor": "$theme.colors.mono400",
  "borderStyle": "solid",
  "borderTopLeftRadius": "1px",
  "borderTopRightRadius": "4px",
  "borderWidth": "1px",
  "boxShadow": "0px 1px 4px rgba(0, 0, 0, 0.12)",
  "cursor": "inherit",
  "display": "flex",
  "height": "24px",
  "justifyContent": "center",
  "width": "12px",
}
`;

exports[`Slider styled components StyledThumb StyledThumb: StyledThumb has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono100",
  "borderBottomLeftRadius": "4px",
  "borderBottomRightRadius": "4px",
  "borderColor": "$theme.colors.mono400",
  "borderStyle": "solid",
  "borderTopLeftRadius": "4px",
  "borderTopRightRadius": "4px",
  "borderWidth": "1px",
  "boxShadow": "0px 1px 4px rgba(0, 0, 0, 0.12)",
  "cursor": "inherit",
  "display": "flex",
  "height": "24px",
  "justifyContent": "center",
  "width": "24px",
}
`;

exports[`Slider styled components StyledTick StyledTick: StyledTick has correct styles 1`] = `
Object {
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
}
`;

exports[`Slider styled components StyledTickBar StyledTickBar: StyledTickBar has correct styles 1`] = `
Object {
  "alignItems": "center",
  "display": "flex",
  "justifyContent": "space-between",
  "paddingBottom": "$theme.sizing.scale400",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
}
`;

exports[`Slider styled components StyledTrack StyledTrack: StyledTrack has correct styles 1`] = `
Object {
  "cursor": "inherit",
  "display": "flex",
  "paddingBottom": "$theme.sizing.scale600",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale1000",
}
`;
